### PR TITLE
feat: MultiHeadedOutput 

### PR DESCRIPTION
**Pull request by supersergiy**

### CODE REVIEW

File: notebook/convblock_new_post.ipynb
Comment (torms3) [inline notebook/convblock_new_post.ipynb:1]: Let's not merge the notebooks, they're the ones I used to verify the fix I made to `ConvBlock`.

File: notebook/convblock_new_pre.ipynb
Comment (torms3) [inline notebook/convblock_new_pre.ipynb:1]: Let's not merge the notebooks, they're the ones I used to verify the fix I made to `ConvBlock`.

File: notebook/convblock_old.ipynb
Comment (torms3) [inline notebook/convblock_old.ipynb:1]: Let's not merge the notebooks, they're the ones I used to verify the fix I made to `ConvBlock`.

**Comment by supersergiy** [comment]
The commit in question makes changes to both architecture and training regime, which I'm trying to keep separate. Let's just merge this and the training PR, and then move on to https://github.com/ZettaAI/zetta_utils/commit/558938ddb2e621d17fa2cde4774411ae65263f18 separately

**Pull request merged**